utils/tfhe-versionable/examples/transparent_then_not.rs

```diff
@@ -0,0 +1,173 @@
+//! This example is similar to the "transparent" one, except that the wrapper type is transparent at
+//! a point in time, then converted into its own type that is not transparent.
+//!
+//! Here we have a type, `MyStructWrapper`, that was a transparent wrapper for `MyStruct` in the v0
+//! and v1 of the application. `MyStruct` has been upgraded between v0 and v1. In v2,
+//! `MyStructWrapper` was transformed into an enum. Since it was transparent before, it has no
+//! history (dispatch enum) before v2.
+//!
+//! To make this work, we consider that the inner and the wrapper type share the same history up to
+//! the version where the transparent attribute has been removed.
+
+use std::convert::Infallible;
+
+use tfhe_versionable::{Unversionize, Upgrade, Version, Versionize, VersionsDispatch};
+
+// This type was transparent before, but it has now been transformed to a full type, for example by
+// adding a new kind of metadata.
+#[derive(Versionize)]
+#[versionize(MyStructWrapperVersions)]
+struct MyStructWrapper<T> {
+    inner: MyStruct<T>,
+    count: u64,
+}
+
+// We need to create a dispatch enum that has the same history as the inner type until the point
+// where the wrapper is not transparent anymore.
```

Comment:
I think its better worded that way, but, wording now makes me understand that the enum should be
```rust
enum MyStructWrapperVersions<T> {
    V0(MyStructV0),
    V1(MyStrucV1<T>),
    V2(MyStructWrapper<T>),
}
```

as for V0 and V1 the Wrapper was transparent, and so the inner types were `MyStructV0` and `MyStrucV1`.

because as it stands,
```rust
enum MyStructWrapperVersions<T> {
    V0(MyStructWrapperV0),
    V1(MyStructWrapperV1<T>),
    V2(MyStructWrapper<T>),
}
```
Does not look different  from what is usally done for non transparent structs

Reply:
What I mean here is that in v0 and v1(versions of the app) the `MyStructWrapperVersions` enum does not exist, so we need to create it directly in v2 with at least the same versions as `MyStructVersions`.

Whereas with normal non transparent types you build your enum one version at a time, so you never create from scratch a dispatch enum with 3 variants

Reply:
the examples in this repo try to follow a pattern where "mod v0", "mod v1"... simulate previous versions of an application to show how the versioning evolve with time

Reply:
I reworded it a bit